src/ppom-storage.ts

```diff
@@ -0,0 +1,204 @@
+import { calculateSHA256 } from './crypto-utils';
+
+type FileInfo = {
+  name: string;
+  chainId: string;
+  version: string;
+  checksum: string;
+};
+
+export type PPOMStorageMetadata = FileInfo[];
+
+export type StorageKey = {
+  name: string;
+  chainId: string;
+};
+
+export type StorageBackend = {
+  read(key: StorageKey): Promise<any>;
+  write(key: StorageKey, data: any): Promise<void>;
+  delete(key: StorageKey): Promise<void>;
+  dir(): Promise<StorageKey[]>;
+};
+
+/**
+ * PPOMStorage class
+ * This class is responsible for managing the local storage
+ * It provides the following functionalities:
+ * 1. Sync the metadata with the version info from the cdn
+ * 2. Read a file from the local storage
+ * 3. Write a file to the local storage
+ *
+ * It also validates the checksum of the file when reading and writing in order to
+ * detect corrupted files or files that are not up to date
+ */
+export class PPOMStorage {
+  readonly #storageBackend: StorageBackend;
+
+  readonly _readMetadata: () => PPOMStorageMetadata;
```

Comment:
Is this intended to be private? If so then what do you think about using `#`? (Same for `_writeMetadata`.)